### DISCUSSION TITLE
Force specific traits / colors?

### DISCUSSION THREAD

**Discussion by cryptovladX** [nftchef/art-engine#274] [Help + Q&A] [answered]
Hi there community, I recently stumbled upon this project and after giving it a few tests it looks really promising, kudos to the developers at Hashlips and to this modified fork by nftchef :)

I am having a little issue or question here so some guidance would be greatly appreciated. Is there a way to enforce the same color being used based on different layers? Let me give you an example, here is a sample structure:

- Hair
  - Yellow.png
  - Black.png
  - Brown.png
- Beard
  - Yellow.png
  - Black.png
  - Brown.png

In an ideal case scenario, I would like to choose same color traits when running the script, meaning that if I have Yellow Hair I should also have a Yellow Beard? I hope this makes sense. It would be greatly appreciated if someone can point me in the right direction, thank you.

**Comment by neilryanh** [comment]
One way I've worked around this is running the logic from color first.
- All Hair
  - Yellow
     - Facial Hair
     - Hair
  - Black
     - Facial Hair
     - Hair
  - Brown
     - Facial Hair
     - Hair

**Comment by cryptovladX** [comment]
That's an awesome suggestion, thank you @neilryanh I get the gist of the idea, but could you show me how this would look like in the actual layerConfigurations? I can't seem to grasp it. Thank you so much, your help is greatly appreciated.

**Comment by neilryanh** [comment, marked as answer]
```
Something like this as an example:

All Hair
¦ 
+---Blue#25
¦   +---Facial Hair
¦   ¦       Side Burns#10.png
¦   ¦       NONE#90.png
¦   ¦       
¦   +---Hair Style
¦		Curly#10.png
¦		Short#10.png
+---Red#25
¦   +---Facial Hair
¦   ¦       Side Burns#10.png
¦   ¦       NONE#90.png
¦   ¦       
¦   +---Hair Style
¦		Curly#10.png
¦		Short#10.png
+---Gray#25
¦   +---Facial Hair
¦   ¦       Side Burns#10.png
¦   ¦       NONE#90.png
¦   ¦       
¦   +---Hair Style
¦		Curly#10.png
¦		Short#10.png
+---Blonde#25
    +---Facial Hair
    ¦       Side Burns#10.png
    ¦       NONE#90.png
    ¦       
    +---Hair Style
		Curly#10.png
		Short#10.png
```
EDIT: Then, in your config, all you call is the top folder 
`{ name: "All Hair" },`

**Comment by cryptovladX** [comment]
Fantastic, thank you so much @neilryanh you're a life-saver, next coffee is on me :)

**Comment by cryptovladX** [comment]
Just tested this method and I can confirm it works! 1000 Thanks again @neilryanh

**Comment by neilryanh** [comment]
Awesome! Happy to help.
